### PR TITLE
cmake: avoid rebuilding extensions, and using python-config

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/cephfs/setup.py [hunk 1]
```diff
@@ -48,37 +48,15 @@ def monkey_with_compiler(compiler):
...
-    }
+def get_python_flags(libs):
+    return dict(
+        include_dirs=[distutils.sysconfig.get_python_inc()],
+        library_dirs=distutils.sysconfig.get_config_vars('LIBDIR', 'LIBPL'),
+        libraries=(libs +
+                   [lib.replace('-l','') for lib in
+                    distutils.sysconfig.get_config_vars('LIBS', 'SYSLIBS').split()]),
+        extra_compile_args=distutils.sysconfig.get_config_var('CFLAGS').split(),
+        extra_link_args=distutils.sysconfig.get_config_vars('LDFLAGS', 'LINKFORSHARED').split())
```
Comment (cbodley) [inline src/pybind/cephfs/setup.py]:
```
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/cephfs/setup.py", line 179, in <module>
    **get_python_flags(['cephfs'])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/pybind/cephfs/setup.py", line 57, in get_python_flags
    distutils.sysconfig.get_config_vars('LIBS', 'SYSLIBS').split()]),
AttributeError: 'list' object has no attribute 'split'
```

**Comment by sebastian-philipp** [comment]
One this is still a bit unfortunate, we still don't have a proper python package for each binding. We're still just building the libraries.

**Comment by tchaikov** [comment]
@sebastian-philipp i don't follow you. what do you mean by "a proper python package"? we build these extensions using `setup.py` with distutils. i think this is the pythonic way to build python extensions.

**Comment by sebastian-philipp** [comment]
@tchaikov yep they are.

**Comment by tchaikov** [comment]
@cbodley the FTBFS should be addressed in the latest change.

**Comment by tchaikov** [comment]
retest this please

**Pull request merged**